src/ui/components/NetworkMonitor/utils.ts

```diff
@@ -30,42 +30,74 @@ export type RequestSummary = {
   start: number;
   status: number;
   time: number;
+  type: RequestType;
   url: string;
 };
 
-export const REQUEST_TYPES = {
+export const CANONICAL_REQUEST_TYPES = [
+  "CSS",
+  "Fetch/XHR",
+  "Font",
+  "HTML",
+  "Image",
+  "Javascript",
+  "Manifest",
+  "Media",
+  "Other",
+  "WASM",
+  "Websocket",
+] as const;
+
+export type CanonicalRequestType = typeof CANONICAL_REQUEST_TYPES[number];
+
+// From https://github.com/RecordReplay/gecko-dev/blob/webreplay-release/devtools/server/actors/replay/network-helpers.jsm#L14
+export const REQUEST_TYPES: Record<string, CanonicalRequestType> = {
+  document: "HTML",
+  subdocument: "HTML",
+  objectSubdoc: "HTML",
+
+  fetch: "Fetch/XHR",
   xhr: "Fetch/XHR",
-  javascript: "Javascript",
-  html: "HTML",
-  css: "CSS",
+
+  beacon: "Other",
+  csp: "Other",
+  dtd: "Other",
+  invalid: "Other",
+  object: "Other",
+  other: "Other",
+  ping: "Other",
+  xslt: "Other",
+
   font: "Font",
   img: "Image",
-  manifest: "Manifest",
+  imageset: "Image",
+  webManifest: "Manifest",
   media: "Media",
-  other: "Other",
+  script: "Javascript",
+  stylesheet: "CSS",
   wasm: "WASM",
   websocket: "Websocket",
 };
 
+export type RequestType = keyof typeof REQUEST_TYPES;
```

Comment:
I'm gonna guess this won't be used anywhere once the types are exported as an enum.